src/canvas/Geometry/ArcCurve.tsx

```diff
@@ -32,8 +33,71 @@ export class ArcCurve extends Curve {
                 this.endAngle += 2 * Math.PI;
             }
         }
+
+        this.counterClockwise = this.startAngle > this.endAngle;
     }
 
+    clone = (): ArcCurve => {
+        return new ArcCurve(this.start, this.end, this.control);
+    };
+
+    // Overrides the abstract method in the parent class.
+    drawTo = (path: Path2D): void => {
+        path.arc(this.center.x, this.center.y, this.radius, this.startAngle, this.endAngle, this.counterClockwise);
+    };
+
+    // Overrides the approximation algorithm in the parent class.
+    getLength = (): number => {
+        return Math.abs(this.startAngle - this.endAngle) * this.radius;
+    };
+
+    getOffsetSegments = (distance: number): Segment[] => {
+        const tangentAtStart = this.getTangent(0);
+        const displacementOfStart = Vector.findOpposite(Vector.findPerpVector(tangentAtStart)).normalize().multiplyByScalar(distance);
+
+        const tangentAtEnd = this.getTangent(1);
+        const displacementOfEnd = Vector.findOpposite(Vector.findPerpVector(tangentAtEnd)).normalize().multiplyByScalar(distance);
+
+        const result = this.clone();
+        result.start = Point.translate(this.start, displacementOfStart);
+        result.end = Point.translate(this.end, displacementOfEnd);
+        result.radius = result.start.distanceTo(result.center);
+        result.control = Point.translate(Point.translate(this.control, displacementOfStart), displacementOfEnd); 
+        // result's center, startAngle and endAngle don't need to be updated.
+        return [result];
+    };
+
+    getTangent = (t: number): Vector => {
+        const x = -1 * (this.endAngle - this.startAngle) * this.radius * Math.sin(this.lerp(this.startAngle, this.endAngle, t));
+        const y = (this.endAngle - this.startAngle) * this.radius * Math.cos(this.lerp(this.startAngle, this.endAngle, t));
+        return new Vector(x, y);
+    };
+
+    reversedClone = (): ArcCurve => {
+        return new ArcCurve(this.end, this.start, this.control);
+    };
+
+    scale = (scalar: number): void => {
+        console.log("arc pre scale", this);
+        this.start = this.start.scale(scalar);
+        this.control = this.control.scale(scalar);
+        this.end = this.end.scale(scalar);
+        this.center = this._computeCenter();
+        this.radius = this.center.distanceTo(this.start);
+        this.startAngle = Vector.vectorBetweenPoints(this.center, this.start).getAngle();
+        this.endAngle = Vector.vectorBetweenPoints(this.center, this.end).getAngle();
+        if (Math.abs(this.endAngle - this.startAngle) > Math.PI) {
+            if (this.startAngle < this.endAngle) {
+                this.startAngle += 2 * Math.PI;
+            } else {
+                this.endAngle += 2 * Math.PI;
+            }
+        }
+        
```

Comment:
That is a good point. I removed this section of code and added a check for negative scalar. Should we do this check in the scale function of all of our geometry?

Reply:
I hadn't thought about other Segments, but good point. Even Point's scale should be checked. But I used a negative scalar to scale vectors in a few places, because it is useful for us to flip vectors sometimes, so not for vectors.

Reply:
I moved the throw error for negative scalar to Segment. I think it only makes sense to check at that level because we are trying to keep the geometry from being flipped.

Reply:
I agree. the new scale() method in Segment.tsx is missing a semi. Other than that it looks ok.

Reply:
I added the semi.